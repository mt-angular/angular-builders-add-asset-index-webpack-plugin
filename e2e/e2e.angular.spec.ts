import { execAsyncCommand, readFileAsync } from './util';
import path from 'path';
import { pathNormalize } from '@upradata/node-util';

const angularDir = path.join(__dirname, pathNormalize('../angular-test'));

describe(
    // tslint:disable-next-line: max-line-length
    'Angular test suite snapshots: angular.json set with architect.build.builder: "./linked_modules/custom-webpack-builder-dist-copy:browser and builder.options.customWebpackConfig.path: "./extra-webpack.config.js"', () => {

        beforeAll(() => {
            jest.setTimeout(30000);
        });

        const configs = [
            { title: 'development build', command: `(cd ${angularDir} && ng build)` },
            { title: 'production build', command: `(cd ${angularDir} && ng build --prod)` }
        ].map(c => [ c.title, c.command ]);

        test.each(configs)(
            'Snapshot index.html: %s', async (title: string, command: string) => {
                await execAsyncCommand(command);

                const indexHTML = await readFileAsync(
                    path.resolve(__dirname, angularDir, pathNormalize('dist/angular-test/index.html')),
                    { encoding: 'utf8' });

                expect(indexHTML).toMatchSnapshot();
            });

    });
