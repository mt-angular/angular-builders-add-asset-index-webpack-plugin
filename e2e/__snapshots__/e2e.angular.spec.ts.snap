// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Angular test suite snapshots: angular.json set with architect.build.builder: "./linked_modules/custom-webpack-builder-dist-copy:browser and builder.options.customWebpackConfig.path: "./extra-webpack.config.js" Snapshot index.html: development build 1`] = `
"<!doctype html>
<html lang=\\"en\\">
<head>
  <meta charset=\\"utf-8\\">
  <title>AngularTest</title>
  <base href=\\"/\\">

  <meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1\\">
  <link rel=\\"icon\\" type=\\"image/x-icon\\" href=\\"favicon.ico\\">
<link href=\\"src/assets/font/libre-franklin/libre-franklin-v2-latin-400.woff2\\"><link href=\\"src/assets/font/rubik/rubik-v7-latin-300.woff2\\"><link href=\\"src/assets/font/rubik/rubik-v7-latin-400.woff2\\"><link href=\\"src/assets/font/rubik/rubik-v7-latin-500.woff2\\"><link href=\\"src/assets/font/rubik/rubik-v7-latin-700.woff2\\"></head>
<body>
  <app-root></app-root>
</body>
</html>
"
`;

exports[`Angular test suite snapshots: angular.json set with architect.build.builder: "./linked_modules/custom-webpack-builder-dist-copy:browser and builder.options.customWebpackConfig.path: "./extra-webpack.config.js" Snapshot index.html: production build 1`] = `
"<!doctype html>
<html lang=\\"en\\">
<head>
  <meta charset=\\"utf-8\\">
  <title>AngularTest</title>
  <base href=\\"/\\">

  <meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1\\">
  <link rel=\\"icon\\" type=\\"image/x-icon\\" href=\\"favicon.ico\\">
<link href=\\"src/assets/font/libre-franklin/libre-franklin-v2-latin-400.8d503c823a91e889c0e7.woff2\\"><link href=\\"src/assets/font/rubik/rubik-v7-latin-300.eb02199844b4e2a0871c.woff2\\"><link href=\\"src/assets/font/rubik/rubik-v7-latin-400.1ecc21093d3c8aa44b3e.woff2\\"><link href=\\"src/assets/font/rubik/rubik-v7-latin-500.d3fd715f234b8dcdd09d.woff2\\"><link href=\\"src/assets/font/rubik/rubik-v7-latin-700.9aff6f99df5790ad5dbd.woff2\\"></head>
<body>
  <app-root></app-root>
</body>
</html>
"
`;
